### DISCUSSION TITLE
Installation Help

### DISCUSSION THREAD

**Discussion by pascal-sauer** [magpiemodel/magpie#650] [Help]
In this discussion you can ask questions and report problems you encountered while setting up MAgPIE using the [how to install](https://github.com/magpiemodel/magpie?tab=readme-ov-file#how-to-install) section of the readme.

### Rtools not found
If you get an error like `Error : Could not find tools necessary to compile a package` on Windows your Rtools setup is not working yet. Make sure you installed the RTools version matching your R version. Make sure the environment variable `RTOOLS43_HOME` (for R/Rtools 4.3; replace "43" with your R/Rtools version if needed) is set to the path to the Rtools directory.

### R package system requirements missing
When installing R packages from source (usually only on Linux/Mac) sometimes system libraries are missing. You can look up system requirements and how they are called for your system at `https://sysreqs.r-hub.io/pkg/<package-name>`, replacing `<package-name>` with the package that fails to install. Then you usually just need to install the missing packages via your package manager, e.g. `brew` on Mac, and `apt` on Ubuntu.

For example for the `ragg` package: https://sysreqs.r-hub.io/pkg/ragg
So on Mac you'd need to run `brew install libpng libtiff`, on Ubuntu it would be `sudo apt install libpng-dev libtiff-dev`.

**Comment by EricZnevergiveup** [comment]
Hi everyone,

I'm encountering an error while trying to run MAgPIE according to the "HOW TO INSTALL" guide, specifically at the "download and run MAgPIE" step when executing `Rscript start.R`.

The installation of `m4fsdp` fails with the following message:

```
- Installing m4fsdp ...                         FAILED
Error: Error installing package 'm4fsdp':
==================================

* installing *source* package 'm4fsdp' ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
Error: .onLoad failed in loadNamespace() for 'units', details:
  call: NULL
  error: no database found!
Execution halted
ERROR: lazy loading failed for package 'm4fsdp'
```

I've already followed the "check setup is complete" steps, and all requirements were successfully found.

Could anyone please help me understand what might be causing this "no database found!" error for the `units` package and how I might resolve it?

Thanks in advance!

**Reply by pascal-sauer** [reply]
Hi Eric, this seems like a problem with your `units` installation, not with m4fsdp. Can you check if `library(units)` throws the same error?
You might be missing a udunits system lib. What system are you working on? On Ubuntu you can install it with `sudo apt-get install libudunits2-dev`, then try to reinstall the units package in R with `install.packages('units')`

**Reply by EricZnevergiveup** [reply]
Hi Pascal,

Thanks for your helpful guidance! I managed to solve the `units` package issue and successfully install `m4fsdp`. Here's what worked for me:

The Problem: The `units` package couldn't find the udunits2 database, causing the "no database found" error during `m4fsdp` installation.

The Solution:

1. Install udunits2 via conda (system-level installation): `conda install -c conda-forge udunits2`
2. Set permanent environment variables in R's `.Renviron` file: `usethis::edit_r_environ()`, and add the path where udunits2 is located
3. Restart RStudio, test `library(units)` and the units package now consistently finds the udunits2 database

The key insight was that temporary environment variable settings (`Sys.setenv()`) only work within the current R session. Using `.Renviron` makes the settings persistent across all R sessions.

After this fix, `m4fsdp` installed smoothly and I can proceed with the magpie setup. Hope this helps others who might face similar issues on Windows!

Thanks again for pointing me in the right direction!

**Reply by pascal-sauer** [reply]
Glad you were able to fix it, and thanks a lot for these detailed steps how you solved it, this might indeed help others!
